services/brig/test/integration/API/User/Auth.hs

```diff
@@ -671,19 +863,24 @@ listCookiesWithLabel b u l = do
 -- | Check that the cookie returned after login is sane.
 --
 -- Doesn't check everything, just some basic properties.
-assertSanePersistentCookie :: Http.Cookie -> Assertion
+assertSanePersistentCookie :: forall u . ZAuth.UserTokenLike u => Http.Cookie -> Assertion
 assertSanePersistentCookie ck = do
     assertBool "type" (cookie_persistent ck)
     assertBool "http-only" (cookie_http_only ck)
     assertBool "expiry" (cookie_expiry_time ck > cookie_creation_time ck)
     assertBool "domain" (cookie_domain ck /= "")
     assertBool "path" (cookie_path ck /= "")
+    let mtoken = (fromByteString (cookie_value ck)) :: Maybe (ZAuth.Token u)
+    assertBool "cookie token exists" (isJust mtoken)
+    let (Just token) = mtoken
+    assertBool "cookie token has correct type" ((token^.ZAuth.header.ZAuth.typ) == ZAuth.zauthType @u)
```

Comment:
very nit-picky stuff, leading to a question that maybe genuinely interesting in the end:

- I feel strongly we should require operators be surrounded spaces in our coding style.  This is a good case to motivate this rule: the double-use of `.` for module qualification and function composition is very hard to read.
- The error message from pattern match failure in line 875 is just as good as the one from the assertion in 874, so this code can be simplified.
- I personally like type information best where a name is bound (ie., introduced).
- Too many parens.  :)

So this could be rewritten:

```haskell
    let Just (token :: ZAuth.Token u) = fromByteString (cookie_value ck)
        tokentype = ZAuth.zauthType @u
    assertBool "cookie token has correct type" $ token ^. ZAuth.header . ZAuth.typ == tokentype
```

...  and then I still have one more question: Can `token` really be parsed if it fails the subsequent assertion?  Should it?